src/components/Timer.vue

```diff
@@ -0,0 +1,373 @@
+            return 0;
+        } else {
+            return 1 - this._timeLeft / this._totalTime;
+        }
+    }
+
+    private get _timeCircleInfo(): CircleInfo {
+        // Have a max length to make it more recognizable that this is a timer by never rendering a full circle.
+        // The rounded stroke ending rendered with radius strokeWidth/2 does not count towards the stroke length,
+        // therefore to get the desired gap of 1.5 strokeWidths, we use 2.5 strokeWidths.
+        const maxLength = this.fullCircleLength - 2.5 * this.strokeWidth;
+        const length = Math.min(maxLength, (1 - this._progress) * this.fullCircleLength);
+        const lengthWithLineCaps = length + this.strokeWidth; // add line caps with strokeWidth/2 radius
+        const gap = this.fullCircleLength - length;
+        // The path grows clockwise starting on the right side. Offset by 90 degrees and gap to let path start with gap
+        // and end on top.
+        const offset = this.fullCircleLength / 4 - gap;
+        return { length, lengthWithLineCaps, gap, offset, strokeWidth: this.strokeWidth };
+    }
+
+    private get _fillerCircleInfo(): CircleInfo {
+        // Filler circle should be rendered in the gap left by the time circle with a margin of strokeWidth. If there
+        // is not enough space, compensate by reducing the filler circle stroke width.
+        const availableSpace = this.fullCircleLength - this._timeCircleInfo.lengthWithLineCaps - 2 * this.strokeWidth;
+        const lengthWithLineCaps = Math.max(0, availableSpace);
+        const strokeWidth = Math.min(this.strokeWidth, lengthWithLineCaps);
+        const length = Math.max(0, lengthWithLineCaps - strokeWidth); // subtract rounded line caps
+        const gap = this.fullCircleLength - length;
+        const offset = this.fullCircleLength / 4 // rotate by 90 degrees
+            - this.strokeWidth / 2 // skip rounded line cap of time circle
+            - this.strokeWidth // margin
+            - strokeWidth / 2; // account for our own line cap
+        return { length, lengthWithLineCaps, gap, offset, strokeWidth };
+    }
+
+    private get _updateInterval(): number {
+        const baseSize = 26;
+        const timerSize = (this.$el as HTMLAnchorElement).offsetWidth || baseSize;
+        const scaleFactor = timerSize / baseSize;
+        const circleLengthPixels = this.fullCircleLength * scaleFactor;
+        const steps = circleLengthPixels * 3; // update every .33 pixel change for smooth transitions
+        const minInterval = 1000 / 60; // up to 60 fps
+        const maxInterval = this.detailsShown && this._timeLeft < 60000
+            ? 500 // when counting down seconds update more regularly
+            : Number.POSITIVE_INFINITY;
+        return Math.max(minInterval, Math.min(maxInterval, this._totalTime / steps));
+    }
+
+    @Watch('detailsShown', { immediate: true })
+    private _setRadius() {
+        this.radius.tweenTo(this.detailsShown ? 12 : 8, 300);
```

Comment:
Do these numbers correspond to something? If so can be added as comment to the constants.

```ts
private static readonly RADIUS_BASE = 8;
private static readonly RADIUS_GROW_FACTOR = 1.5;
```
```suggestion
        this.radius.tweenTo(Timer.RADIUS_BASE * (this.detailsShown ? Timer.RADIUS_GROW_FACTOR : 1), 300);
```